src/servicehandler.cpp

```diff
@@ -194,181 +92,236 @@ ServiceHandler::ServiceHandler()
+            // invitationhandlerinterface doesn't have error-function
+            // This is utterly broken by design, except when the invitation plugin is also a service plugin
+            ServiceInterface *s = service(notebook.pluginName());
+            if (!s) {
+                s = service(defaultName);
+            }
+            d->mError = s ? (ServiceHandler::ErrorCode) s->error() : ErrorInternal;
+            return false;
+        }
+    } else {
         return false;
-
-    return d->executePlugin(SendInvitation, invitation, body, calendar, storage, notebook);
+    }
 }
 
 
-bool ServiceHandler::sendUpdate(const Incidence::Ptr &invitation, const QString &body,
-                                const ExtendedCalendar::Ptr &calendar, const ExtendedStorage::Ptr &storage,
-                                const Notebook::Ptr &notebook)
+bool ServiceHandler::sendUpdate(const Notebook &notebook, const Incidence &invitation, const QString &body)
 {
-    if (storage.isNull() || invitation.isNull() || calendar.isNull())
+    InvitationHandlerInterface* plugin = d->invitationPlugin(notebook.pluginName());
+
+    d->mError = ErrorOk;
+    if (plugin) {
+        if (plugin->sendUpdate(notebook.account(), invitation, body)) {
+            return true;
+        } else {
+            // service needed to get possible error, because
+            // invitationhandlerinterface doesn't have error-function
+            // This is utterly broken by design, except when the invitation plugin is also a service plugin
+            ServiceInterface *s = service(notebook.pluginName());
+            if (!s) {
+                s = service(defaultName);
+            }
+            d->mError = s ? (ServiceHandler::ErrorCode) s->error() : ErrorInternal;
+            return false;
+        }
+    } else {
         return false;
-
-    return d->executePlugin(SendUpdate, invitation, body, calendar, storage, notebook);
+    }
 }
 
 
-bool ServiceHandler::sendResponse(const Incidence::Ptr &invitation, const QString &body,
-                                  const ExtendedCalendar::Ptr &calendar, const ExtendedStorage::Ptr &storage,
-                                  const Notebook::Ptr &notebook)
+bool ServiceHandler::sendResponse(const Notebook &notebook, const Incidence &invitation, const QString &body)
 {
-    if (storage.isNull() || invitation.isNull() || calendar.isNull())
+    InvitationHandlerInterface* plugin = d->invitationPlugin(notebook.pluginName());
+
+    d->mError = ErrorOk;
+    if (plugin) {
+        if (plugin->sendResponse(notebook.account(), invitation, body)) {
+            return true;
+        } else {
+            // service needed to get possible error, because
+            // invitationhandlerinterface doesn't have error-function
+            // This is utterly broken by design, except when the invitation plugin is also a service plugin
+            ServiceInterface *s = service(notebook.pluginName());
```

Comment:
Could prefer to avoid single character variables unless they are loop iterators or such.

Reply:
Ah sure, I've put a more sensible variable name.